File: code/csharp/DesignPatterns/chainofresponsibility/LevelOneSupport.cs
public class LevelOneSupport : ISupportLevel
{
    private ISupportLevel _next;

    public void SetNext(ISupportLevel next)
    {
        _next = next;
        Console.WriteLine($"Level one support setting next level to {next.GetType().Name} Support");
    }

    public void HandleRequest(SupportTicket ticket)
    {
        if (ticket.Level <= 1)
        {
            Console.WriteLine($"Level One Support handling ticket: {ticket.Description}");
            Console.WriteLine("Ticket resolved.");
        }
        else
        {
            Console.WriteLine($"Level One Support cannot handle ticket: {ticket.Description} Escalating to next level.");
            _next.HandleRequest(ticket);
            Console.WriteLine("Ticket returned to level one after being resolved.");
        }
    }
}

File: code/typescript/chainofresponsibility/LevelOneSupport.ts
import { ISupportLevel } from "./ISupportLevel";
import { SupportTicket } from "./SupportTicket";

export class LevelOneSupport implements ISupportLevel {
    private next: ISupportLevel;

    setNext(next: ISupportLevel): void {
        this.next = next;
        console.log(`Level one support setting next level to ${next.constructor.name} Support`);
    }

    handleRequest(ticket: SupportTicket): void {
        if (ticket.level <= 1) {
            console.log(`Level One Support handling ticket: ${ticket.description}`);
            console.log("Ticket resolved.");
        } else {
            console.log(`Level One Support cannot handle ticket: ${ticket.description} Escalating to next level.`);
            this.next.handleRequest(ticket);
            console.log("Ticket returned to level one after being resolved.");
        }
    }
}
